lib/iris/_lazy_data.py

```diff
@@ -536,9 +536,12 @@ def lazy_elementwise(lazy_array, elementwise_op):
     # This makes good practical sense for unit conversions, as a Unit.convert
     # call may cast to float, or not, depending on unit equality : Thus, it's
     # much safer to get udunits to decide that for us.
-    dtype = elementwise_op(np.zeros(1, lazy_array.dtype)).dtype
+    meta = da.utils.meta_from_array(lazy_array)
+    new_meta = elementwise_op(meta)
```

Comment:
This now requires `elementwise_op` to work on an empty-shaped array, which was not the case before, and may not work for all array operations.
So I think that passing a minimal array of the correct dtype, as before, would be "safer".

Reply:
I restored the previous behaviour in c1e5e1a